src/Evolution/Systems/GrMhd/ValenciaDivClean/FiniteDifference/ReconstructWork.tpp

```diff
@@ -218,35 +240,66 @@ void reconstruct_prims_work(
     DirectionMap<3, gsl::span<const double>> ghost_cell_vars{};
 
     for (const auto& direction : Direction<3>::all_directions()) {
+      DirectionalId<3> id{};
       if (element.neighbors().contains(direction)) {
         const auto& neighbors_in_direction = element.neighbors().at(direction);
         ASSERT(neighbors_in_direction.size() == 1,
                "Currently only support one neighbor in each direction, but "
                "got "
                    << neighbors_in_direction.size() << " in direction "
                    << direction);
-        ghost_cell_vars[direction] =
-            gsl::make_span(get<tag>(neighbor_data.at(DirectionalId<3>{
-                               direction, *neighbors_in_direction.begin()}))[0]
-                               .data(),
-                           number_of_variables * neighbor_num_pts);
+        id = DirectionalId<3>{direction, *neighbors_in_direction.begin()};
       } else {
         // retrieve boundary ghost data from neighbor_data
         ASSERT(
             element.external_boundaries().count(direction) == 1,
-            "Element has neither neighbor nor external boundary to direction : "
+            "Element has neither neighbor nor external boundary to direction: "
                 << direction);
-        ghost_cell_vars[direction] = gsl::make_span(
-            get<tag>(neighbor_data.at(DirectionalId<3>{
-                direction, ElementId<3>::external_boundary_id()}))[0]
-                .data(),
-            number_of_variables * neighbor_num_pts);
+        id = DirectionalId<3>{direction, ElementId<3>::external_boundary_id()};
       }
+      if constexpr (std::is_same_v<tag, hydro::Tags::Temperature<DataVector>>) {
+        ASSERT(number_of_variables == 1,
+               "Should only have one tensor component for a Scalar");
+        if (reconstruct_density_times_temperature) {
+          ghost_cell_vars[direction] = gsl::make_span(
+              &buffer_for_recons_vars[volume_num_pts +
+                                      (2 * direction.dimension() +
+                                       (direction.side() == Side::Upper ? 1
+                                                                        : 0)) *
+                                          neighbor_num_pts],
+              number_of_variables * neighbor_num_pts);
+
+          const auto& data_in_dir = neighbor_data.at(id);
+          DataVector view{
+              /// NOLINTNEXTLINE(cppcoreguidelines-pro-type-const-cast)
+              const_cast<double*>(ghost_cell_vars[direction].data()),
```

Comment:
Do you need the const_cast here?  It looks like the pointer isn't const, and if it is assigning to it in the next expression seems fishy.

Reply:
I see, I missed that.  But the new version is better, anyway.